mcp-server/coneshare_mcp/tools/documents.py

```
@@ -47,3 +48,70 @@ async def delete_document(
+        path: Optional[str] = Field(default=None, description="Optional root-relative destination folder path (e.g. 'finance/2026')"),
+        content_type: str = Field(default="application/octet-stream", description="MIME type of the file (e.g., 'application/pdf')"),
+    ) -> dict:
+        """Upload a small or medium-sized document (under 20 MB) directly using base64 file content in a single step.
+
+        Use this tool when uploading text files, small PDFs, images, or documents under 20 MB.
+        For large files (> 20 MB), use request_document_upload and finalize_document_upload instead.
+        """
+        client = ConeshareClient.from_ctx(ctx)
+        try:
+            file_bytes = base64.b64decode(content_base64)
+        except Exception as exc:
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)